### DISCUSSION TITLE
Save screenshot for every simulation frame

### DISCUSSION THREAD

**Discussion by Junang-Wang** [sofa-framework/sofa#4961] [Write / Run a simulation] [answered]
Hi there,
I need to run a series of parametrized simulations in sofa from a server without a screen. I am wondering if it's possible to save a screenshot for every simulation frame without launching the sofa GUI. I know I can do that from the sofa GUI by pressing "V." But as I said, I am runing Sofa from a server.

Thanks for advance

**Comment by alxbilger** [comment, marked as answer]
Here are some solutions:

- Use a GUI and the Python function [`SaveScreenshot`](https://sofapython3.readthedocs.io/en/latest/content/modules/Sofa/generated/Sofa.Gui/classes/Sofa.Gui.GUIManager.html#Sofa.Gui.GUIManager.SaveScreenshot) (it does not work in batch)
- Use [Headlessrecorder](https://github.com/sofa-framework/sofa/blob/master/Sofa/GUI/HeadlessRecorder/README.md), but it is being deprecated: https://github.com/sofa-framework/sofa/pull/4928
- Use [OffscreenCamera](https://github.com/jnbrunet/SofaOffscreenCamera), but it is not really supported (probably out of date)
- Offline rendering: save your meshes and load them in Blender, then render.

**Reply by Junang-Wang** [reply]
Thank you for your help
